common/src/main/kotlin/com/gitlab/kordlib/common/entity/DiscordGuild.kt

```diff
@@ -234,28 +237,60 @@ data class DiscordVoiceState(
         val suppress: Boolean
 )
 
+/**
+ * Nitro boosting tier of a Guild.
+ *
+ * @property level the boost level of the guild.
+ * @property maxEmotes Maximal amount of server emotes. Source: https://support.discord.com/hc/en-us/articles/360028038352-Server-Boosting-#h_419c3bd5-addd-4989-b7cf-c7957ef92583
+ */
 @Serializable(with = PremiumTier.PremiumTierSerializer::class)
-enum class PremiumTier(val level: Int) {
+sealed class PremiumTier {
+    abstract val level: Int?
+    abstract val maxEmotes: Int?
+
     /** The default code for unknown values. */
-    Unknown(Int.MIN_VALUE),
-    None(0),
-    One(1),
-    Two(2),
-    Three(3);
+    object Unknown : PremiumTier() {
```

Comment:
I've come to the realization that every premium tier unknown to us would have a `level`, so making it `null` is nonsensical.
Since we know the level from the decoder, we can turn `Unknown` into a class with the level in its constructor instead, and remove the nullable type marker from the parent sealed class.

Reply:
What about the maxEmotes then? surely we won't have access to them

Reply:
Indeed, those should stay nullable.